vectors/no_resizable_vec.rs

```diff
@@ -223,23 +224,69 @@ impl<T> IndexMut<usize> for NoResizableVec<T> {
     }
 }
 
-impl<T> BorshSerialize for NoResizableVec<T> {
-    fn serialize<W: Write>(&self, _writer: &mut W) -> borsh::maybestd::io::Result<()> {
-        Ok(())
+impl<T: AnchorSerialize> BorshSerialize for NoResizableVec<T> {
+    fn serialize<W: Write>(&self, writer: &mut W) -> borsh::maybestd::io::Result<()> {
+        let len = usize::try_from(self.len()).map_err(|_| std::io::ErrorKind::InvalidInput)?;
+        writer.write_all(&len.to_le_bytes())?;
+
+        let cap = usize::try_from(self.buf.cap).map_err(|_| std::io::ErrorKind::InvalidInput)?;
+        writer.write_all(&cap.to_le_bytes())?;
+
+        // serialize the vector
+        unsafe {
+            let slice = core::slice::from_raw_parts(self.buf.ptr.as_ptr(), self.len);
+            slice.serialize(writer)
+        }
     }
 }
 
 /// We need to fix the capacity of the vector.
 /// However, this number depends on the specific verification task.
-impl<T:Nondet> BorshDeserialize for NoResizableVec<T> {
-    fn deserialize(_buf: &mut &[u8]) -> borsh::maybestd::io::Result<Self> {
-        let res = NoResizableVec::new(10) ;
-        Ok(res)
+impl<T:Nondet + AnchorDeserialize> BorshDeserialize for NoResizableVec<T> {
+    fn deserialize(buf: &mut &[u8]) -> borsh::maybestd::io::Result<Self> {
+        // Deserialize the len and capacity
+        let len = usize::deserialize(buf)?;
+        let capacity = usize::deserialize(buf)?;
+
+        // Deserialize the vector
+        Vec::deserialize(buf).map(|vec: Vec<T>| {
```

Comment:
we should not use `Vec` in the definition of `NoResizableVec` since this will bring back re-allocations that `Vec` might be doing. Rather inline the call to `deserialize` and augment as appropriate.